### PR TITLE
feat(server-portal): create ConfigurationLoader

### PR DESCRIPTION
**Pull request by Tzal3x**
A utility class that makes safer to load the environment
variables of the project.

By using this class, we can ensure that the environment
variables are loaded correctly and their data types are
correct.

### CODE REVIEW

File: portal/server/configuration-loader.ts
Comment (giac-mysten) [inline portal/server/configuration-loader.ts:1]: Nit: the name should be `configuration_loader.ts`

File: portal/server/configuration-loader.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    portalDomainNameLength?: number;
+    premiumRpcUrlList: string[];
+    rpcUrlList: string[];
+    enableSentry: Boolean;
+    sentryAuthToken?: string;
+    sentryDsn?: string;
+}
+
+/**
+* A utility class that makes safer to load the environment
```
Comment (giac-mysten) [inline portal/server/configuration-loader.ts:29]: Nit:
```suggestion
* A utility class that makes it safe to load the environment
```
and also line length

File: portal/server/configuration-loader.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+type StringBoolean = 'true' | 'false';
+function isStringBoolean(value: string): value is StringBoolean {
+    return value === 'true' || value === 'false';
+}
+function toBoolean(value: string): Boolean {
+    return value === 'true';
+}
```
Comment (giac-mysten) [inline portal/server/configuration-loader.ts:10]: Question: I guess there is no way to store a private yaml file or similar instead of env variables?

**Review by giac-mysten — APPROVED** [review]
Awesome stuff :)

File: portal/server/configuration-loader.ts (hunk 2 above)
Comment (Tzal3x) [reply portal/server/configuration-loader.ts:10]: Yea there is no such built-in functionality in Vercel. Other custom options (such as using vercel's blob store to save yaml files, and then load them on the portal) seemed to overcomplicate the logic.

**Pull request merged**